src/main/java/htsjdk/samtools/cram/encoding/huffman/codec/CanonicalHuffmanIntegerEncoding.java

```diff
@@ -28,12 +28,12 @@
 import java.nio.ByteBuffer;
 import java.util.Arrays;
 import java.util.Map;
+import java.util.Objects;
 
 public class CanonicalHuffmanIntegerEncoding implements Encoding<Integer> {
     private static final EncodingID ENCODING_ID = EncodingID.HUFFMAN;
     private int[] bitLengths;
     private int[] values;
-    private final ByteBuffer buf = ByteBuffer.allocate(1024 * 10);
```

Comment:
Moved inside method where it belongs, and it shouldn't be used for equality tests

Reply:
I don't think you want to localize this to the method it's used in, because that means re-allocating it every time.  That seems like it could be a substantial performance hit through increased garbage collection.  

 You can just ignore it in equality checks.

Reply:
good point - I'll fix that in the general Encoding refactor as well